core/validatorapi/eth2types.go

```diff
@@ -85,21 +84,42 @@ type proposerDutiesResponse struct {
 	Data          []*eth2v1.ProposerDuty `json:"data"`
 }
 
+// proposeBlockResponsePhase0 defines the phase0 Version response to the proposeBlock endpoint
+// see https://ethereum.github.io/beacon-APIs/#/Validator/produceBlockV2
 type proposeBlockResponsePhase0 struct {
-	Version spec.DataVersion    `json:"version"`
+	Version string              `json:"version"`
 	Data    *eth2p0.BeaconBlock `json:"data"`
 }
 
+func (p *proposeBlockResponsePhase0) Init(data *eth2p0.BeaconBlock) {
```

Comment:
Not a fan of mutable APIs. Remember functions over methods and immutable over mutable.